pkg/kn/commands/service/service.go

```diff
@@ -36,3 +43,16 @@ func NewServiceCommand(p *commands.KnParams) *cobra.Command {
 	serviceCmd.AddCommand(NewServiceUpdateCommand(p))
 	return serviceCmd
 }
+
+func waitForService(client serving_kn_v1alpha1.KnClient, serviceName string, out io.Writer, timeout int) error {
+	fmt.Fprintf(out, "Waiting for service '%s' to become ready ... ", serviceName)
```

Comment:
And what happens if user `CTRL-C` out. I assume the API request continues but `kn` fails with user cancel? Might want to be clear. Since alternatively you could trap `CTL-C` and try to cancel cleanly or even delete service. Not advocating for clean up at this point but users will want to cancel and let's anticipate that and aid them.

Reply:
Good point. But I would interpret CTL-C as "cancelling the waiting" (that is where it currently states). We could try to catch that and print out something like "Cancelled waiting for service to become ready". It then just turns into an async behaviour where you have to check the server state. 

But a complete rollback is imo out of scope (that would mean to implement a fully transactional behaviour, recording the previous state and as you are on the client you don't have full control what's happening on the server-side. At least some events have been produced, which might trigger some other, non-rollbackable actions etc. A client-side managed transaction with possible compensation actions is not something easy.

Reply:
OK on canceling the waiting. Best.

Reply:
> Since alternatively you could trap CTL-C and try to cancel cleanly or even undo change
this would be a very surprising behaviour to most users.

Reply:
I still have to add that kind of trapping for CTRL-C and printing out something like

"Interrupt: Stop waiting on service completion. Please check the service's status yourself."

or so.